File: live-coding/live-coding-master/long_repeat/solutions/longRepeat.cs
public static int longRepeat(String text) {

    int longCount = 0;
    char curChar = '';
    int curCount = 0;

    /*

    Students might also use regular for loop, e.g.:

    for (int i = 0; i < text.Length; i++) {
    }

    In which case they'll need to access the current character
    via the [] index operator or similar technique.

    */


    foreach (char letter in text)
    {

        if (letter == curChar)
        {
            curCount++;
        }
        else
        {
            curChar = char;
            curCount = 1;
        }

        if curCount > longCount {
            longCount = curCount;
        }

    }

    return longCount;

}

File: live-coding/live-coding-master/long_repeat/solutions/long_repeat.py
def long_repeat(line):

    long_count = 0
    cur_char = ""
    cur_count = 0

    """
    Students might also use a range for loop, e.g.:

    for in in range(len(text)):

    In which case they'll need to access the current character
    via the [] index operator.
    """

    for char in line:

        if char == cur_char:
            cur_count += 1
        else:
            cur_char = char
            cur_count = 1
        
        if cur_count > long_count:
            long_count = cur_count
    
    return long_count